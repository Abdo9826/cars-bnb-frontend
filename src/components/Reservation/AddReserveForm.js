/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { useForm } from 'react-hook-form';
import { useSelector } from 'react-redux';
// import { useParams } from 'react-router-dom';

const AddReserveForm = () => {
  const state = useSelector((state) => state);
  const { cars, user } = state;
  console.log(user);
  console.log(cars);
  const params = {};
  // console.log(params);
  const currentCar = cars.cars.find((car) => car.id === Number(params.carId)) || {};
  const { register, handleSubmit, formState: { errors } } = useForm({
    defaultValues: {
      city: '',
      date: '',
      car_id: '',
      user_id: '',
    },
  });

  const onSubmit = (data, e) => {
    e.preventDefault();
    const car = { ...data, car_id: currentCar.id, user_id: user.id };
    console.log(car);
    console.log(data);
  };

  return (
    <div>
      <form onSubmit={handleSubmit(onSubmit)}>
        <input
          type="text"
          placeholder="City"
          {...register('city', { required: true })}
        />
        {errors.city && <span>City is required</span>}
        <input
          type="date"
          name="date"
          placeholder="Date"
          {...register('date', { required: true })}
        />
        {errors.date && <span>Date is required</span>}
        { cars.addFromNav ? (
          <select
            name="car_id"
            {...register('car_id', { required: true })}
          >
            {cars.cars.map((car) => (
              <option key={car.id} value={car.id}>
                {car.model}
              </option>
            ))}
          </select>
        ) : (
          <select
            name="car_id"
            {...register('car_id', { required: true })}
          >
            <option key={currentCar.id} value={currentCar.id}>
              {currentCar.model}
            </option>
          </select>
        )}
        <input type="submit" value="Reserve" />
      </form>
    </div>
  );
};

export default AddReserveForm;
