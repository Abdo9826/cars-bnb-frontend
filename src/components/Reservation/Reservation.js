import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { deleteReserve, fetchReservations } from '../../redux/Reservations/ReserveSlice';

function Reservations() {
  const dispatch = useDispatch();
  const reservations = useSelector((state) => state.reservations.reservations) || [];
  const cars = useSelector((state) => state.cars.cars);

  useEffect(() => {
    dispatch(fetchReservations());
  }, [dispatch]);

  return (
    <div className="container">
      <div className="row">
        <div className="col-md-12">
          <h1 className="text-center">Reservations</h1>
          <div className="row">
            {reservations.length <= 0 && (
              <div className="loading">
                <h4 className="text-center">No reservations</h4>
              </div>
            )}
            {reservations.map((reservation) => (
              <div className="col-md-4" key={reservation?.id}>
                <div className="card mb-4">
                  <div className="card-body">
                    <img
                      src={
                        cars.find((car) => car.id === reservation.car_id)?.featured_image
                      }
                      alt="car"
                      className="img-fluid"
                    />
                    {reservation?.car_id ? (
                      <h5 className="card-title">
                        {
                          cars.find((car) => car.id === reservation.car_id)?.model
                        }
                      </h5>
                    ) : (
                      <h5 className="card-title">Car not found</h5>
                    )}

                    <p className="card-text">{reservation?.city}</p>
                    <p className="card-text">{reservation?.reservation_date}</p>

                    <button
                      type="button"
                      onClick={() => dispatch(deleteReserve(reservation?.id))}
                      className="btn btn-danger"
                    >
                      Remove
                    </button>
                  </div>
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
}

export default Reservations;
