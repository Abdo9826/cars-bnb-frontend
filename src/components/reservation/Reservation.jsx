import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

function Reservations() {
  const reservations = useSelector((state) => state.reservations.reservations);

  console.log(reservations);

  return (
    <div>
      <h1>My reservations</h1>
      <ul>
        {reservations.map((reservation) => (
          <li key={reservation.id}>
            <Link to={`/reservations/${reservation.id}`}>
              {reservation.reservation_date}
              {reservation.updated_at}
              {reservation.city}
              {reservation.user_id}
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default Reservations;
