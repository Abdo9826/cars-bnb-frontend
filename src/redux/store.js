import { configureStore } from '@reduxjs/toolkit';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import logger from 'redux-logger';
import CarsReducer from './Cars/carSlice';
import authenticationReducer from './Auth/authenticationSlice';
import ReserveReducer from './Reservations/ReserveSlice';
import reservationListReducer from './reservationList/reservationListSlice';

export default configureStore({
  reducer: {
    user: authenticationReducer,
    cars: CarsReducer,
    addReservations: ReserveReducer,
    reservations: reservationListReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});
