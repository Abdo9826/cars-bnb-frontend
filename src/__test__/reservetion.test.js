/* eslint-disable */
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Reservation from '../components/reservation/Reservation';
import store from '../redux/store';

it('To check test is working', () => {
  expect(true).toBe(true);
});

it('Test if header render correctly', () => {
  const TREE = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Reservation />
      </BrowserRouter>
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});
