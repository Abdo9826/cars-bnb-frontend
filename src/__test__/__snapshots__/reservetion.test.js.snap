// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if header render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <h1
        className="text-center"
      >
        Reservations
      </h1>
      <div
        className="row"
      >
        <div
          className="loading"
        >
          <h4
            className="text-center"
          >
            No reservations
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
